test/zx.test.mjs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {assert, test as t} from './test-utils.mjs'
+
+const test = t.bind(null, 'zx')
+
+if (test('supports `-v` flag / prints version')) {
+  let v = (await $`node zx.mjs -v`).toString().trim()
+  assert.equal(v, require('../package.json').version)
+}
+
+if (test('prints help')) {
+  let help
+  try {
+    await $`node zx.mjs`
+  } catch (err) {
+    help = err.toString().trim()
+  }
+  assert(help.includes('print current zx version'))
+}
+
+if (test('supports `--experimental` flag')) {
+  await $`echo 'echo("test")' | node zx.mjs --experimental`
+}
+
+if (test('supports `--quiet` flag / Quiet mode is working')) {
+  let p = await $`node zx.mjs --quiet docs/markdown.md`
+  assert(!p.stdout.includes('whoami'))
+}
+
+if (test('Eval script from https ref')) {
+  let script = path.resolve('test/fixtures/echo.http')
+  let server = quiet($`while true; do cat ${script} | nc -l 8080; done`)
```

Comment:
With timeout? Problem we can’t merge them

Reply:
Missing Content-Length. Fixed.

>   let script = `
  await question('foo or bar? ', {choices: ['foo', 'bar']})
`
  $`node zx.mjs <<< ${script}`
  
  ```
foo or bar? file:///Users/antongolub/projects/zx/src/index.mjs:99
      let output = new ProcessOutput({
                   ^
ProcessOutput [Error]: 
    at file:///Users/antongolub/projects/zx/test/index.test.mjs:123:4
    exit code: 13
    at ChildProcess.<anonymous> (file:///Users/antongolub/projects/zx/src/index.mjs:99:20)
    at ChildProcess.emit (node:events:520:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)

  ```

Still raises 13 err code 13. Seems like a deadlock detection https://stackoverflow.com/questions/46914025/node-exits-without-error-and-doesnt-await-promise-event-callback/46916601#46916601